operations/configuration_management/Troubleshoot_CFS_Session_Failing_to_Complete.md

```diff
@@ -1,29 +1,41 @@
 # Troubleshoot CFS Session Failing to Complete
 
-Troubleshoot issues where Configuration Framework Service \(CFS\) sessions/pods fail and Ansible hangs. These issues can be resolved
-by modifying Ansible to produce less output.
+Troubleshoot issues where Ansible hangs.
 
 ## Prerequisites
 
 A CFS session or pod is failing to complete, and the Ansible logs are not showing progress or completion.
 
-The following is an example of the error causing Ansible to hang:
+## Check for filesystem issues
 
-```text
-PLAY [Compute] *****************************************************************
-META: ran handlers
-META: ran handlers
-META: ran handlers
+Hung sessions are usually a result of filesystem issues, such as problems with DVS, on the nodes that CFS is attempting to configure.
+An issue on even one of the nodes that a session is attempting to configure can cause the whole session to hang (unless Ansible is specifically configured to use a `free` rather than `linear` strategy).
 
-PLAY [Compute] *****************************************************************
-Using module file /usr/lib/python3.6/site-packages/ansible/modules/system/setup.py
-Pipelining is enabled.
-<x5000c3s3b0n1> ESTABLISH SSH CONNECTION FOR USER: root
-<x5000c3s3b0n1> SSH: EXEC ssh -vvv -o ServerAliveInterval=30 -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o 'IdentityFile="/secret/key"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="root"' -o ConnectTimeout=10 -o ControlPath=/root/.ansible/cp/f6f378183d x5000c3s3b0n1 '/bin/sh -c '"'"'/usr/bin/python3 && sleep 0'"'"''
-```
+1. (`ncn-mw#`) Find all the nodes that the Ansible session is targeting.
+
+    In most cases this can be done by looking at the `limit` parameter for the session.
+
+    * When starting with a pod name, run:
+
+        ```bash
+       kubectl -n services describe pod $POD_NAME | grep -A 5 ANSIBLE_ARGS
+       ```
+
+    * When starting with a session name, run:
+
+        ```bash
+       cray cfs sessions describe $SESSION_NAME | jq .ansible.limit
+       ```
```

Comment:
Indentation is inconsistent here.
````suggestion
       ```bash
       kubectl -n services describe pod $POD_NAME | grep -A 5 ANSIBLE_ARGS
       ```

    * When starting with a session name, run:

       ```bash
       cray cfs sessions describe "$SESSION_NAME" --format json | jq .ansible.limit
       ```
````